### PR TITLE
feat: improve puptoo parser logging

### PR DESCRIPTION
**Pull request by SteveHNH**
When a parser fails in system_profile we need to be able to easily
track down which one caused the problem. This new system will log each
parser when it fails and cause the payload to be rejected. We will be
able to search these failures with the request_id in kibana which will
help speed up troubleshooting time.

Signed-off-by: Stephen Adams <tsadams@gmail.com>

### CODE REVIEW

File: src/puptoo/utils/puptoo_logging.py [hunk 1]
```diff
@@ -10,6 +11,9 @@
 from . import config
 
 
+threadctx = local()
```
Comment (beav) [inline src/puptoo/utils/puptoo_logging.py:14]: there's only ever one thread, correct?

File: src/puptoo/utils/puptoo_logging.py [hunk 2]
```diff
@@ -49,3 +54,24 @@ def initialize_logging():
         logger.info("Skipping cloudwatch configuration")
 
     return logger
+
+
+class ContextualFilter(logging.Filter):
+    """
+    This filter gets the request_id from the message and adds it to
+    each log record. This way we do not have to explicitly retreive/pass
+    around the request_id for each log message
```
Comment (beav) [inline src/puptoo/utils/puptoo_logging.py:63]: this is a cool way to do it! I always had to pass request_id around and it was messy.

**Comment by beav** [comment]
The parser has some stuff that could eventually be factored out, but overall I think this is a huge improvement and worth merging. Ack!

**Review by beav — APPROVED** [review]

File: src/puptoo/utils/puptoo_logging.py (hunk 1 above)
Comment (SteveHNH) [reply src/puptoo/utils/puptoo_logging.py:14]: yep!

**Pull request merged**